lib/cpp/src/thrift/server/TNonblockingServer.cpp

```diff
@@ -1393,9 +1397,42 @@ bool TNonblockingIOThread::notify(TNonblockingServer::TConnection* conn) {
     return false;
   }
 
-  const int kSize = sizeof(conn);
-  if (send(fd, const_cast_sockopt(&conn), kSize, 0) != kSize) {
-    return false;
+  fd_set wfds, efds;
+  int ret = -1;
+  int kSize = sizeof(conn);
+  const char * pos = (const char *)const_cast_sockopt(&conn);
+
+  while (kSize > 0) {
+    FD_ZERO(&wfds);
+    FD_ZERO(&efds);
+    FD_SET(fd, &wfds);
+    FD_SET(fd, &efds);
+    ret = select(fd + 1, NULL, &wfds, &efds, NULL);
```

Comment:
Seems like a dangerous assumption that the file descriptor you are interested is +1 from the obtained file descriptor.  Did you mean to use `getNotificationRecvFD()` instead of `fd + 1` here?